### ISSUE TITLE
Invalid examples in Modelica.Media

### ISSUE THREAD

**Issue by modelica-trac-importer** [modelica/Modelica#539]
**Reported by kurzbach on 27 Apr 2011 07:02 UTC**
There are some examples within the media library, where ` medium ` is declared as ` Medium.BaseProperties ` which is a model, and then directly used as an argument to functions, which take a ` ThermoDynamicState ` as argument. 

This is related to:
```
Modelica.Media.Examples.SimpleLiquidWater
Modelica.Media.Examples.TestOnly.MixIdealGasAir
Modelica.Media.Examples.TestOnly.FlueGas
Modelica.Media.Examples.TestOnly.IdealGasN2
Modelica.Media.Examples.TestOnly.IdealGasN2Mix
```

----

Migrated-From: https://trac.modelica.org/Modelica/ticket/539

**Comment by modelica-trac-importer** [comment]
**Modified by kurzbach on 27 Apr 2011 07:37 UTC**

**Comment by modelica-trac-importer** [comment]
**Comment by hubertus on 27 Apr 2011 07:40 UTC**
I will fix that. I had noted that for a few examples, and was totally surprised that it was accepted without throwing an error.

**Comment by modelica-trac-importer** [comment]
**Comment by hubertus on 1 May 2011 21:41 UTC**
Fixed in trunk in 77ff33ec0aa737004854953888b72538adfa5c19.

**Comment by modelica-trac-importer** [comment]
**Comment by hubertus on 1 May 2011 21:51 UTC**
Fixed in 3.1 maintenance in 3bfffa03bc56aa11293e4ea0625d2efa262b0ee7

**Comment by modelica-trac-importer** [comment]
**Comment by kurzbach on 2 May 2011 06:55 UTC**
In the definition of Modelica.Media.Examples.TestOnly.MixIdealGasAir there is still the line:
```
Medium.SpecificEnthalpy h_is = Medium.isentropicEnthalpyApproximation(2.0e5, medium);
```
which should be:
```
Medium.SpecificEnthalpy h_is = Medium.isentropicEnthalpyApproximation(2.0e5, medium.state);
```

**Comment by modelica-trac-importer** [comment]
**Comment by hubertus on 2 May 2011 10:32 UTC**
Fixed remainder in e526b0f3247ab850691f28815f2d1d402aa3f25a, 3bfffa03bc56aa11293e4ea0625d2efa262b0ee7 in trunk and 3.1 maintenance
